### DISCUSSION TITLE
[Q&A]  authentication required, no credentials available

### DISCUSSION THREAD

**Discussion by sebtech69** [glpi-project/glpi-agent#621] [Q&A] [answered]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

i can't get glpiagent to work
my conf :
GLPI 10.0.10, debian12, mariadb, apache
SSO -> active directory
glpiinventory 1.3.4 installed
HTTPS certificate signed by our internal certificate authority
administration -> inventory -> "activate inventory" check

Everything works fine, except the inventory! 
my first question is, what should be the URL to contact? i've seen on the documentation that if the glpiinventrory pluggin is installed i have to use "https://******.****.***/plugins/glpiinventory/" i've also seen other url like "https://******.****.***/glpi/front/inventory.php" or "https://******.****.***/front/inventory.php".)(I tried the three urls for the same result )

here's how I install the glpi-agent (1.7) on my desktops:
msiexec.exe /i GLPI-Agent-1.7-x64.msi /quiet ADD_FIREWALL_EXCEPTION=1 DEBUG=2 AGENTMONITOR=1 RUNNOW=1 EXECMODE=1 NO_SSL_CHECK=1 QUICKINSTALL=1 SERVER=https://******.****.***/plugins/glpiinventory/

after installing the agent i find this in the logs :
[error] [http client] authentication required, no credentials available
[error] No supported answer from server at https://******.****.***/plugins/glpiinventory/

if I add credentials(USER=***** PASSWORD=******* )during installation I get :
[debug] [http client] authentication required, submitting credentials
[error] [http client] authentication required, wrong credentials
[error] No supported answer from server at https://******.****.***/plugins/glpiinventory/

I can't find what the problem is, if anyone could please help me.

**Comment by g-bougard** [comment]
Hi @sebtech69 

you have explanations for the right URL to use in the documentation here, read carefully the Caution note: https://glpi-agent.readthedocs.io/en/latest/configuration.html#server

If you installed glpiinventory via the marketplace, your URL is probably wrong and the agent reach a generic page which ask for authentication.

**Comment by sebtech69** [comment]
thx g-bougard for your help,

I installed glpiinventory manually in / pluggins, i think my url is good and is "https://../plugins/glpiinventory/""

**Reply by sebtech69** [reply]
i tried with glpi native directory and have same issue.
My problème is my SSO, if i desactivate it inventory working.

but I can't manage to exclude this url from the sso, I've tried several techniques but none of them work.

tried  :
1

`#AliasMatch` "^/(plugins/glpiinventory/(index\.php)?)$" "/var/www/glpi/$1"

#<DirectoryMatch  "/var/www/glpi/plugins/glpiinventory">
   #AuthType None
    #Options Indexes FollowSymLinks
    #AllowOverride All
    #Require all granted
    #Satisfy Any
#</DirectoryMatch>
`

2

`#<DirectoryMatch` "/plugins/glpiinventory/">
#AuthType None
#Require all granted
#Satisfy Any
#</DirectoryMatch>
`

3

`#<Directory` /var/www/glpi/plugins/glpiinventory>
#GssapiBasicAuth off
#AuthType None
#Require all granted
#Order allow,deny
#llow from all
#</Directory>
`
4

`<If` "! %{REQUEST_URI} =~ /glpiinventory(.*)$/">
#GssapiBasicAuth off
#AuthType None
#Require all granted
#Satisfy Any
#</If>
`

**Reply by g-bougard** [reply, marked as answer]
Hi @sebtech69 

sorry but your post in simply understandable, you have to reformat the content and select the "code" format for every part needing to be output as raw and don't include dash in front of the line if it's not a commented line.

**Reply by sebtech69** [reply]
sorry better like that
